pkg/providers/vsphere/vsphere.go

```diff
@@ -915,6 +915,54 @@ func (p *vsphereProvider) SetupAndValidateUpgradeCluster(ctx context.Context, cl
 	if err != nil {
 		return fmt.Errorf("failed setup and validations: %v", err)
 	}
+	err = p.validateMachineConfigsNameUniqueness(ctx, cluster, clusterSpec)
+	if err != nil {
+		return fmt.Errorf("failed validate machineconfig uniqueness: %v", err)
+	}
+	return nil
+}
+
+func (p *vsphereProvider) validateMachineConfigsNameUniqueness(ctx context.Context, cluster *types.Cluster, clusterSpec *cluster.Spec) error {
+	prevSpec, err := p.providerKubectlClient.GetEksaCluster(ctx, cluster, clusterSpec.GetName())
+	if err != nil {
+		return err
+	}
+
+	cpmc := clusterSpec.Spec.ControlPlaneConfiguration.MachineGroupRef.Name
```

Comment:
nit: personal opinion, this kind of acronyms make the code way more difficult to read. I believe using thing like `capi` is ok because it's a shared term in the context of the project. But this one is a bit more obscure. Specially in this case because it's a name and not an object per se

Reply:
yeah I did find it kind of hard to read myself tbh